CB-5089 Move tez's jar location to cloudstorage

cegganesh84 opened the pull request:
1. Right now this jar is stored in HDFS using CM command https://github.infra.cloudera.com/Starship/cmf/blob/cdpd-master/web/src/main/java/com/cloudera/cmf/service/tez/TezUploadTarCommand.java#L84
    2. Adding a safety valve for tez.lib.uris property to set the location to a cloudstorage.
    3. The directory structure follows what is in CM already.
    4. We need to change the CM code to use this property instead of the HDFS location hardcoding. Once that is promoted we can promote this change.
    5. Have the tez jar isolated into datahub location, to prevent issues arising from concurrent creation of datahub clusters.
    6. Moved to Junit5.
    
    ./gradlew clean build

lajosrodek commented on template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/tez/TezRoleConfigProviderTest.java:
```diff
@@ -1,12 +1,16 @@
+import com.sequenceiq.cloudbreak.cloud.model.ClouderaManagerRepo;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.junit.MockitoJUnitRunner;
```
I know most legacy unit tests are still based on JUnit 4, but we shall gradually migrate them to JUnit 5 if possible. Could you please use the new Jupiter types here? See `HiveKnoxConfigProviderTest` as an example.

cegganesh84 replied:
Addressed!

lajosrodek commented on template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/tez/TezRoleConfigProviderTest.java:
```diff
@@ -1,12 +1,16 @@
 
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.core.CoreConfigProvider.CORE_DEFAULTFS;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
```
Minor: I suggest replacing standard JUnit assertions with AssertJ's `Assertions.assertThat()` because the latter offers a fluent API and more easily readable. There are several examples in this module, e.g. `HiveKnoxConfigProviderTest`.

cegganesh84 replied:
Addressed!

cegganesh84 commented:
Thanks, @lajosrodek for the valuable comments! Will address these.

cegganesh84 commented on template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/tez/TezRoleConfigProvider.java:
```diff
@@ -22,16 +26,53 @@
+            if (cdh.isEmpty()) {
+                return "";
+            }
+            return cdh.get().getVersion() + "/";
```
Addressed!

cegganesh84 commented on template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/tez/TezRoleConfigProvider.java:
```diff
@@ -1,17 +1,21 @@
+
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils.config;
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.core.CoreConfigProvider.CORE_DEFAULTFS;
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.hive.HiveMetastoreCloudStorageServiceConfigProvider.HMS_METASTORE_EXTERNAL_DIR;
```
Addressed!

cegganesh84 commented:
@lajosrodek I have addressed your comments. Also marked it do not merge for now. Figured out that the right dependency order is CM change first, then the CB change.

doktoric commented:
@lajosrodek if you think this still needed then please merge

cegganesh84 commented:
Will close this for now. The related CM PR was not merged.
